File: validate_json.py

import json
import sys

file_path = 'D:/WebsiteBaru/YourChords2.0/lib/data/guitar.json'

try:
    with open(file_path, 'r', encoding='utf-8') as f:
        data = json.load(f)
    print("VALID_JSON")
except json.JSONDecodeError as e:
    print(f"INVALID_JSON: {e.msg}")
    print(f"Line: {e.lineno}")
    print(f"Column: {e.colno}")
    print(f"Pos: {e.pos}")
    
    # Print context
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        if e.lineno <= len(lines):
            print(f"Content at line {e.lineno}:")
            print(repr(lines[e.lineno - 1]))
except Exception as e:
    print(f"ERROR: {e}")

File: validate_json.js

const fs = require('fs');

try {
    const data = fs.readFileSync('D:/WebsiteBaru/YourChords2.0/lib/data/guitar.json', 'utf8');
    JSON.parse(data);
    console.log("JSON is valid");
} catch (e) {
    console.log("JSON is invalid");
    console.log(e.message);
    if (e.message.includes('position')) {
        const match = e.message.match(/position (\d+)/);
        if (match) {
            const pos = parseInt(match[1]);
            const start = Math.max(0, pos - 50);
            const end = Math.min(data.length, pos + 50);
            console.log("Context:");
            console.log(data.substring(start, end));
            console.log("Pointer:");
            console.log(" ".repeat(pos - start) + "^");
        }
    }
}
